From b48fe3ba22e95ed7c14a58155b6b142cbb88757f Mon Sep 17 00:00:00 2001
From: Nate Graham <nate@kde.org>
Date: Wed, 17 Dec 2025 09:07:11 -0700
Subject: [PATCH] Hide "reboot after update" setting for SteamOS

Like the RPM-OSTree and Systemd-Sysupdate backends, it always requires a
reboot after update.
---
 kcm/CMakeLists.txt | 4 ++++
 kcm/updates.cpp    | 3 ++-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/kcm/CMakeLists.txt b/kcm/CMakeLists.txt
index a3c895f7c..fd4472ddb 100644
--- a/kcm/CMakeLists.txt
+++ b/kcm/CMakeLists.txt
@@ -26,6 +26,10 @@ target_link_libraries(kcm_updates PRIVATE
     KF6::KCMUtilsQuick
 )
 
+if (BUILD_SteamOSBackend)
+    target_compile_definitions(kcm_updates PRIVATE WITH_STEAMOS_BACKEND)
+endif()
+
 if (BUILD_RpmOstreeBackend)
     target_compile_definitions(kcm_updates PRIVATE WITH_OSTREE_BACKEND)
 endif()
diff --git a/kcm/updates.cpp b/kcm/updates.cpp
index 28c4eb68e..3ccdbc435 100644
--- a/kcm/updates.cpp
+++ b/kcm/updates.cpp
@@ -40,7 +40,8 @@ DiscoverSettings *Updates::discoverSettings() const
 
 bool Updates::mandatoryRebootAfterUpdate() const
 {
-#if defined WITH_SYSUPDATE_BACKEND
+#if defined(WITH_SYSUPDATE_BACKEND) || defined(WITH_STEAMOS_BACKEND)
+    return true;
     return true;
 #elif defined WITH_OSTREE_BACKEND
     return QFile::exists(QStringLiteral("/run/ostree-booted"));
-- 
GitLab

